front/lib/api/assistant/actions/process.ts

```diff
@@ -0,0 +1,416 @@
+import type {
+  AgentActionSpecification,
+  AgentConfigurationType,
+  AgentMessageType,
+  ConversationType,
+  ModelId,
+  ModelMessageType,
+  ProcessActionType,
+  ProcessConfigurationType,
+  ProcessErrorEvent,
+  ProcessParamsEvent,
+  ProcessSuccessEvent,
+  Result,
+  TimeFrame,
+  UserMessageType,
+} from "@dust-tt/types";
+import {
+  cloneBaseConfig,
+  DustProdActionRegistry,
+  Ok,
+  renderSchemaPropertiesAsJSONSchema,
+} from "@dust-tt/types";
+
+import { runActionStreamed } from "@app/lib/actions/server";
+import {
+  parseTimeFrame,
+  retrievalAutoTimeFrameInputSpecification,
+  timeFrameFromNow,
+} from "@app/lib/api/assistant/actions/retrieval";
+import { constructPrompt } from "@app/lib/api/assistant/generation";
+import { getSupportedModelConfig } from "@app/lib/assistant";
+import type { Authenticator } from "@app/lib/auth";
+import { AgentProcessAction } from "@app/lib/models/assistant/actions/process";
+import logger from "@app/logger/logger";
+
+/**
+ * Model rendering of process actions.
+ */
+
+export function renderProcessActionForModel(
+  action: ProcessActionType
+): ModelMessageType {
+  let content = "";
+  if (action.outputs === null) {
+    throw new Error(
+      "Output not set on process action; this usually means the process action is not finished."
+    );
+  }
+
+  content += "PROCESSED OUTPUTS:\n";
+
+  // add schema
```

Comment:
is this like a TODO ?

Reply:
Yes marking it as a todo to figure out if we want to add the schema here

Reply:
Could we write "TODO" then so it's clear ?